src/main/java/edu/umich/eecs/april/apriltag/ApriltagDetectorActivity.java

```diff
@@ -69,63 +81,21 @@ protected void onCreate(Bundle savedInstanceState) {
                     new String[]{Manifest.permission.CAMERA},
                     MY_PERMISSIONS_REQUEST_CAMERA);
         } else {
-            this.has_camera_permissions = 1;
+            startCamera();
         }
     }
 
-    /** Release the camera when the application is exited */
     @Override
     protected void onDestroy() {
         super.onDestroy();
-        stopThreads();
-        Log.i(TAG, "Finished destroying.");
-    }
-
-    /** Release the camera when application focus is lost */
-    protected void onPause() {
-        super.onPause();
-        stopThreads();
-        Log.i(TAG, "Finished pause.");
+        detectorExecutor.shutdown();
     }
 
-    private void stopThreads() {
-        if (mCameraPreviewThread != null) {
-            mCameraPreviewThread.interrupt();
-            mCameraPreviewThread.destroy();
-            try {
-                mCameraPreviewThread.join();
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            }
-            mCameraPreviewThread = null;
-        }
-        if (mDetectionThread != null) {
-            mDetectionThread.interrupt();
-            mDetectionThread.destroy();
-            try {
-                mDetectionThread.join();
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            }
-            mDetectionThread = null;
-        }
-    }
-
-    /** (Re-)initialize the camera */
-    protected void onResume() {
```

Comment:
If we don't need to override anything, there's no need to define the hook

Reply:
We need to pause the camera capture, which is not done in the parent class. This is necessary to avoid the camera remaining open when the app goes to the background, or to a lesser extent when a user goes to the settings activity.

Additionally, we need to re-initialize apriltag with new settings, which is not done in the parent class onResume() and this PR removes the reinit from the onResume callback. This PR makes the reinit happen on settings apply, but settings apply doesn't pause the preview callback and apriltag_init is not thread-safe.

Reply:
There's a bit of misunderstanding here. The camerax framework manages the camera lifecycle and pauses the camera when the app is paused. We do not need a separate hook as we did with the raw Camera api.